daml/splice-util-token-standard-wallet/daml/Splice/Util/Token/Wallet/BatchTransferUtility.daml

```diff
@@ -0,0 +1,98 @@
+-- | A utility contract that can be used to provide batch transfer functionality.
+module Splice.Util.Token.Wallet.BatchTransferUtility where
+
+import DA.Map qualified as M
+import DA.Optional (fromOptional)
+
+
+import Splice.Api.Token.HoldingV1 as Api.Token.HoldingV1
+import Splice.Api.Token.TransferInstructionV1
+
+-- | A call to a token standard factory to initiate a token standard transfer.
+data TransferFactoryCall = TransferFactoryCall with
+    factoryCid : ContractId TransferFactory
+    choiceArg : TransferFactory_Transfer
+  deriving (Eq, Show)
+
+-- | Determine the sender of the transfer in the first call.
+getFirstSender : [TransferFactoryCall] -> Party
+getFirstSender [] = error "Cannot determine sender from empty list of TransferFactoryCalls"
+getFirstSender (call :: _) = call.choiceArg.transfer.sender
+
+-- | Utility contract maintained by an operator for third-parties to perform
+-- multiple token standard transfers in a single transaction.
+--
+-- Users of this utility contract should inject it to their command submissions
+-- using explicit disclosure.
+template BatchTransferUtility
+  with
+    operator: Party
+    -- FIXME: consider whether to add a userWhitelist : [Party]
+  where
+    signatory operator
+
+    -- A choice for a sender to execute many token standard transfers in a single transaction.
+    nonconsuming choice BatchTransferUtility_ExecuteBatch : BatchTransferUtility_ExecuteBatchResult
```

Comment:
Question: is this enough batching?

I'd expect so as it allows to also smuggle in self-transfers if required.